### PR TITLE
docs: add more consistent table schema in 300-many-to-many-relations.mdx

### PR DESCRIPTION
**Pull request by joaobispo2077**
## Describe this PR

While a friend (@Xula) was reading the relationship documentation he found an SQL that didn't conform to a Prisma schema, he tell me. So I test the examples from docs, adapted this SQL and also added an expandable example of the real SQL generated in a migration from that Prisma schema.

## Changes

- Refactored SQL to fit more in Prisma schema.
- Added expandable example of real SQL generated from Prisma schema.

### CODE REVIEW

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 1]
```diff
@@ -71,6 +73,45 @@ CREATE TABLE "CategoriesOnPosts" (
 CREATE UNIQUE INDEX "CategoriesOnPosts_category_post_unique" ON "CategoriesOnPosts"("categoryId" int4_ops,"postId" int4_ops);
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: Interesting that this Unique index seems to be missing on the real, generated SQL (see below). Could point to a logic error here.

Comment (joaobispo2077) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: Good point, thanks.

@Xula @janpio 
Is this example below more accurately?

```sql
CREATE TABLE "Category" (
    id SERIAL PRIMARY KEY
    name TEXT NOT NULL
);

CREATE TABLE "Post" (
    id SERIAL PRIMARY KEY
    title TEXT NOT NULL
);

CREATE TABLE "CategoriesOnPosts" (
    "categoryId" INTEGER NOT NULL,
    "postId" INTEGER NOT NULL,
    "assignedBy" TEXT NOT NULL,    
    "assignedAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
    FOREIGN KEY ("categoryId") REFERENCES "Category"("id") ON DELETE RESTRICT ON UPDATE CASCADE,
    FOREIGN KEY ("postId") REFERENCES "Post"("id") ON DELETE RESTRICT ON UPDATE CASCADE,
    CONSTRAINT "CategoriesOnPosts_pkey" PRIMARY KEY ("postId", "categoryId")
);
```

Comment (Xula) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: @joaobispo2077 I think yes!

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 2]
```diff
@@ -55,22 +55,65 @@ The underlying SQL looks like this:
 ```sql
 CREATE TABLE "Category" (
     id SERIAL PRIMARY KEY
+    name TEXT NOT NULL
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:58]: This I agree with 👍

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 3]
```diff
@@ -55,22 +55,65 @@ The underlying SQL looks like this:
...
 );
+
 CREATE TABLE "Post" (
     id SERIAL PRIMARY KEY
+    title TEXT NOT NULL
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:63]: This I agree with 👍

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 4]
```diff
@@ -55,22 +55,65 @@ The underlying SQL looks like this:
...
+-- Relation table  -------------------------------------------------------
+CREATE TABLE "CategoriesOnPosts" (
+    "categoryId" INTEGER NOT NULL,
+    "postId" INTEGER NOT NULL,
+    "assignedBy" TEXT NOT NULL,    
+    "assignedAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    FOREIGN KEY ("categoryId") REFERENCES "Category"("id") ON DELETE RESTRICT ON UPDATE CASCADE,
+    FOREIGN KEY ("postId") REFERENCES "Post"("id") ON DELETE RESTRICT ON UPDATE CASCADE,
+    CONSTRAINT "CategoriesOnPosts_pkey" PRIMARY KEY ("postId", "categoryId")
+);
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:75]: I agree with the capitalization changes here.

But why the change from Unique index to PK?
Why the addition of the referential actions?
(Both might be justified, I just want to ahve the explicit reason to understand.)

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 5]
```diff
@@ -55,22 +55,65 @@ The underlying SQL looks like this:
...
+
+    CONSTRAINT "CategoriesOnPosts_pkey" PRIMARY KEY ("postId","categoryId")
 );
-CREATE UNIQUE INDEX "CategoriesOnPosts_category_post_unique" ON "CategoriesOnPosts"("categoryId" int4_ops,"postId" int4_ops);
+
+-- AddForeignKey
+ALTER TABLE "CategoriesOnPosts" ADD CONSTRAINT "CategoriesOnPosts_postId_fkey" FOREIGN KEY ("postId") REFERENCES "Post"("id") ON DELETE RESTRICT ON UPDATE CASCADE;
+
+-- AddForeignKey
+ALTER TABLE "CategoriesOnPosts" ADD CONSTRAINT "CategoriesOnPosts_categoryId_fkey" FOREIGN KEY ("categoryId") REFERENCES "Category"("id") ON DELETE RESTRICT ON UPDATE CASCADE;
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:111]: This was created explicitly using Prisma migrate with the schema above?

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 6]
```diff
@@ -55,22 +55,65 @@ The underlying SQL looks like this:
...
+    FOREIGN KEY ("categoryId") REFERENCES "Category"("id") ON DELETE RESTRICT ON UPDATE CASCADE,
+    FOREIGN KEY ("postId") REFERENCES "Post"("id") ON DELETE RESTRICT ON UPDATE CASCADE,
+    CONSTRAINT "CategoriesOnPosts_pkey" PRIMARY KEY ("postId", "categoryId")
+);
+```
+
+<details><summary>Expand for a migration SQL that is generated from this schema</summary>
+
+```sql
+-- CreateTable
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:81]: This unfortunately confuses our renderer:
image
@tanberry @nikolasburk  Do we have a way to fix/avoid this?

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 2 above)
Comment (tomhoule) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:58]: +1

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 3 above)
Comment (tomhoule) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:63]: +1

**Review by tomhoule — COMMENTED** [review]
I understand the change now, and same diagnostic and questions as @janpio

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 6 above)
Comment (tanberry) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:81]: Odd that when you look at Preview deploy of this page, it results in only one hyphen (- CreateTable) in this first instance, but in subsequent instances we have two hyphens (-- CreateTable).

Is this what you are commenting on, @janpio ?

Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx:81]: The correct content is double hyphen.
Our renderer understands the first one as an indication to format the content as a deletion in a diff (see the red background color on the line in the screenshot). That should not happen.

**Comment by nikolasburk** [comment]
Hey @joaobispo2077 👋 thanks so much for taking the time to improve our docs 💚 

Since there have been some clarifying questions but no responses, I'm closing this PR due to inactivity. If you feel strongly about it, feel free to re-open it any time and share some thoughts about the questions by @janpio so that we can look into this again 🙏
